src/xdist/dsession.py

```diff
@@ -118,11 +118,15 @@ def pytest_runtestloop(self):
         assert self.sched is not None
 
         self.shouldstop = False
+        pending_exception = None
         while not self.session_finished:
             self.loop_once()
             if self.shouldstop:
-                self.triggershutdown()
-                raise Interrupted(str(self.shouldstop))
+                if not self.shuttingdown:
```

Comment:
Perhaps it would be safer/less error prone to move this `if not self.shuttingdown` to `triggershutdown()`, so it becomes a no-op if `self.shuttingdown` is already True?

Reply:
I like that. I'll make the change.

Reply:
Bit puzzled about the change log entry. Is 'improvement' supposed to be the $type in the name of the changelog file? I ask because it is not one of the defined types, if my understanding is correct,

Reply:
Yeah, something like `changelog/963.improvement.rst` . 👍

Reply:
OK. Done.